pkg/domain/stateDiffDoguConfig.go

```diff
@@ -0,0 +1,44 @@
+package domain
+
+import (
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/common"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/ecosystem"
+)
+
+func determineDoguConfigDiffs(
+	config map[common.SimpleDoguName]CombinedDoguConfig,
+	actualDoguConfig map[common.DoguConfigKey]ecosystem.DoguConfigEntry,
+) DoguConfigDiff {
+	var doguConfigDiff []DoguConfigEntryDiff
+
+	for _, doguConfig := range config {
+		// present entries
+		for key, expectedValue := range doguConfig.Config.Present {
+			actualEntry, actualExists := actualDoguConfig[key]
+			doguConfigDiff = append(doguConfigDiff, determineDoguConfigDiff(key, string(actualEntry.Value), actualExists, string(expectedValue), true))
+		}
+		// absent entries
+		for _, key := range doguConfig.Config.Absent {
+			actualEntry, actualExists := actualDoguConfig[key]
+			doguConfigDiff = append(doguConfigDiff, determineDoguConfigDiff(key, string(actualEntry.Value), actualExists, string(actualEntry.Value), false))
+		}
+	}
+	return doguConfigDiff
+}
+
+func determineDoguConfigDiff(key common.DoguConfigKey, actualValue string, actualExists bool, expectedValue string, expectedExists bool) DoguConfigEntryDiff {
```

Comment:
🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>func `determineDoguConfigDiff` is unused